dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,207 @@
+            elif not item:
+                return None
+            parent_id = item["id"]
+        return item
+
+    def get_path_id_from_cache(self, path_info):
+        file_id = ""
+        parts = path_info.path.split("/")
+        if parts and (parts[0] in self.cached_root_dirs):
+            parent_id = self.cached_root_dirs[parts[0]]
+            file_id = self.cached_root_dirs[parts[0]]
+            parts.pop(0)
+        else:
+            parent_id = path_info.netloc
+        return file_id, parent_id, parts
+
+    def get_path_id(self, path_info, create=False):
+        file_id, parent_id, parts = self.get_path_id_from_cache(path_info)
+
+        if not parts and file_id:
+            return file_id
+
+        file1 = self.resolve_remote_file(parent_id, parts, create)
+        return file1["id"] if file1 else ""
+
+    def exists(self, path_info):
+        return self.get_path_id(path_info) != ""
+
+    def batch_exists(self, path_infos, callback):
+        results = []
+        for path_info in path_infos:
+            results.append(self.exists(path_info))
+            callback.update(str(path_info))
+        return results
+
+    def _upload(self, from_file, to_info, name, no_progress_bar):
+        dirname = to_info.parent
+        if dirname:
+            parent_id = self.get_path_id(dirname, True)
+        else:
+            parent_id = to_info.netloc
+
+        file1 = self.drive.CreateFile(
+            {"title": to_info.name, "parents": [{"id": parent_id}]}
+        )
+
+        from_file = open(from_file, "rb")
+        if not no_progress_bar:
+            from_file = TrackFileReadProgress(name, from_file)
+
+        file1.content = from_file
+
+        file1.Upload()
+        from_file.close()
+
+    def _download(self, from_info, to_file, name, no_progress_bar):
+        from dvc.progress import Tqdm
+
+        file_id = self.get_path_id(from_info)
+        gdrive_file = self.drive.CreateFile({"id": file_id})
+        if not no_progress_bar:
+            tqdm = Tqdm(desc=name, total=int(gdrive_file["fileSize"]))
+        gdrive_file.GetContentFile(to_file)
+        if not no_progress_bar:
+            tqdm.close()
+
+    def get_file_checksum(self, path_info):
+        raise NotImplementedError
+
+    def list_cache_paths(self):
+        file_id = self.get_path_id(self.path_info)
+        prefix = self.path_info.path
+        for path in self.list_path(file_id):
+            yield prefix + "/" + path
+
+    def walk(self, path_info):
+        raise NotImplementedError
+
+    def list_file_path(self, drive_file):
+        if drive_file["mimeType"] == self.FOLDER_MIME_TYPE:
+            for i in self.list_path(drive_file["id"]):
+                yield drive_file["title"] + "/" + i
+        else:
+            yield drive_file["title"]
+
+    def list_path(self, parent_id):
+        for file1 in self.list_drive_item(
+            "'%s' in parents and trashed=false" % parent_id
```

Comment:
We try to not use `%` notation, and use `format` instead.
```suggestion
            "'{}' in parents and trashed=false".format(parent_id)
```